src/analysis/normed_space/real_inner_product.lean

```diff
@@ -221,6 +222,48 @@ instance inner_product_space_is_normed_space : normed_space ℝ α :=
 
 end norm
 
+section instances
+/-- The standard Euclidean space, `fin n → ℝ`. -/
+def euclidean_space (n : ℕ) : Type := fin n → ℝ
```

Comment:
Note that we use `(n : Type*) [fintype n]` for matrices. Should we use `def euclidean_space (n : Type*) [fintype n]` here?

Reply:
This way we'll immediately get, e.g., Cauchy-Schwarz for finite sums (not only over `range n`).

Reply:
For Cauchy-Schwarz, I'd rather give the direct proof (but for all `L^p` norms). Here, I think `euclidean_space n` is really just an example, and a model space for manifolds. If we see a situation where the fintype situation is needed, it will be easy to refactor.

Reply:
Cauchy-Schwarz / Hölder / Minkowski inequalities probably need to be given in several forms anyway (both integral versions and versions for finite sums; at present we just have Cauchy-Schwarz for inner product spaces, but not any variant of the others, I think).

Reply:
I pushed this and some further simplifications of the proofs. Feel free to (partially) revert.